pkg/tlsconfig/config.go

```diff
@@ -0,0 +1,97 @@
+// Package tlsconfig provides primitives to retrieve secure-enough TLS configurations for both clients and servers.
+//
+// As a reminder from https://golang.org/pkg/crypto/tls/#Config:
+//	A Config structure is used to configure a TLS client or server. After one has been passed to a TLS function it must not be modified.
+//	A Config may be reused; the tls package will also not modify it.
+package tlsconfig
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
+
+// NewConfig returns a secure-enough-by-default TLS configuration object.
+func New() *tls.Config {
+	return &tls.Config{
+		// Avoid fallback to SSL protocols < TLS1.0
+		MinVersion:               tls.VersionTLS10,
+		PreferServerCipherSuites: true,
+		CipherSuites: []uint16{
+			tls.TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,
+			tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+			tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+			tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+			tls.TLS_RSA_WITH_AES_128_CBC_SHA,
+			tls.TLS_RSA_WITH_AES_256_CBC_SHA,
+		},
+	}
+}
+
+// certPool returns an X.509 certificate pool from `ca`, the certificate file.
+func certPool(ca string) (*x509.CertPool, error) {
+	// If we should verify the server, we need to load a trusted ca
+	certPool := x509.NewCertPool()
+	file, err := ioutil.ReadFile(ca)
+	if err != nil {
+		return nil, fmt.Errorf("Could not read CA certificate %s: %v", ca, err)
+	}
+	certPool.AppendCertsFromPEM(file)
+	return certPool, nil
+}
+
+// Client returns a TLS configuration meant to be used by a client.
+func Client(enable, verify bool, ca, cert, key string) (*tls.Config, error) {
+	if !enable {
```

Comment:
@LK4D4 @tiborvass I agree with both of you. `bool` as function arguments are a broken pattern. The meaning of `enable` and `verify` have a space that might not be valid for all values of a TLS configuration. The meaning of the strings `ca, `cert` and `key` are also unclear (paths? raw data? what format?). This is not a good, generalized interface and needs work. We may want an option-style, _al a carte_ function signature, or something else.

On the other, it is arguable whether that is the aim of the PR. Much of the above criticism adds to the complexity but doesn't attack the goal. At least this interface demonstrates the problem and it might be problematic to create a better interface to communicate `verify` and `enable`. At least, we now have this in a single spot.